library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -614,6 +662,7 @@ protected void replay(long windowId)
                 emit(line);
               }
               if (recoveryEntry.fileClosed) {
+                emitEndFileControlTuple(false);
```

Comment:
If file was closed in the window, we emit endBatch CT. false indicates that entire file processing completed. (A file which encounters failure midway while processing will emit control tuple with status true)